src/main/java/ua/edu/internship/interview/data/documents/UserQuestionDocument.java

```diff
@@ -0,0 +1,39 @@
+package ua.edu.internship.interview.data.documents;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import org.bson.types.ObjectId;
+import org.springframework.data.annotation.Id;
+import org.springframework.data.mongodb.core.mapping.Document;
+import org.springframework.data.mongodb.core.mapping.DocumentReference;
+import org.springframework.data.mongodb.core.mapping.Field;
+import ua.edu.internship.interview.data.enumeration.QuestionDifficulty;
+import ua.edu.internship.interview.data.enumeration.QuestionType;
+
+@NoArgsConstructor
+@AllArgsConstructor
+@EqualsAndHashCode(onlyExplicitlyIncluded = true)
+@Getter
+@Setter
+@Builder
+@Document(collection = "user_questions")
+public class UserQuestionDocument {
+    @Id
+    @EqualsAndHashCode.Include
+    @Field(name = "id")
+    private ObjectId id;
+    @Field(name = "userId")
+    private String userId;
+    @DocumentReference
+    private SkillDocument skill;
```

Comment:
One skill can have multiple questions. Maybe it would be better if we extracted the question text, difficulty, type into an embedded document, and UserQuestionDocument will store list of those documents for every skill? Or should we leave it as it is?